activerecord/test/cases/autosave_association_test.rb

```diff
@@ -1574,6 +1574,14 @@ def test_should_automatically_save_bang_the_associated_model
     assert_equal "The Vile Serpent", @pirate.reload.ship.name
   end
 
+  def test_should_automatically_save_bang_the_associated_model_if_it_sets_the_inverse_record
+    pirate = Pirate.new(catchphrase: "Savvy?")
+    ship = Ship.new(name: "Black Pearl")
+    ship.pirate = pirate
+    pirate.save!
+    assert_equal "Black Pearl", pirate.reload.ship.name
```

Comment:
Similar to https://github.com/rails/rails/issues/52332#issuecomment-2245522368.

Without patch:
<img width="1049" alt="Screenshot 2024-07-25 at 12 17 01 am" src="https://github.com/user-attachments/assets/a1506661-b273-45b6-a485-6db9878e57a0">